### PR TITLE
Hot fix from QA feedback

### PR DESCRIPTION
**Pull request by tahmidrahman-dsi**
**Bug from QA**

bug

**Fix after this change**

fix

### CODE REVIEW

**Comment by tahmidrahman-dsi** [comment]
**Test reports**

- Unit tests
image

- E2E tests
image (1)

File: packages/client/src/applications/index.ts [hunk 1]
```diff
@@ -578,6 +579,37 @@ export function mergeDeclaredApplications(
     })
 
   applications.push(...transformedDeclaredApplications)
+  const applicationOrderComparator = function(
```
Comment (mushrafulhoque-dsi) [inline packages/client/src/applications/index.ts:582]: Do we really need it ?

File: packages/client/src/applications/index.ts [hunk 2]
```diff
@@ -586,14 +618,10 @@ async function updateFieldAgentDeclaredApplicationsByUser(
   const state = getState()
   const scope = getScope(state)
 
-  if (
-    !state.applicationsState.applications ||
-    state.applicationsState.applications.length !== 0 ||
-    !scope ||
-    !scope.includes('declare')
-  ) {
+  if (!scope || !scope.includes('declare')) {
```
Comment (mushrafulhoque-dsi) [inline packages/client/src/applications/index.ts:621]: I don't think, we should remove the length check here. It means we will be forcing field agents to pull application from elastic everytime they log in.

File: packages/client/src/views/FieldAgentHome/FieldAgentHome.test.tsx [hunk 3]
```diff
@@ -78,7 +78,7 @@ const countQueryGraphqlMock = {
 merge(mockUserResponse, nameObj)
 
 describe('FieldAgentHome tests', () => {
-  const { store } = createStore()
+  const { store, history } = createStore()
```
Comment (mushrafulhoque-dsi) [inline packages/client/src/views/FieldAgentHome/FieldAgentHome.test.tsx:81]: As we have used the timeout  function now, so we don't need to change it anymore. Can you please revert it ?

**Review by mushrafulhoque-dsi — CHANGES_REQUESTED** [review]
@tahmidrahman-dsi left few comments for you

File: packages/client/src/views/FieldAgentHome/FieldAgentHome.test.tsx (hunk 3 above)
Comment (tahmidrahman-dsi) [reply packages/client/src/views/FieldAgentHome/FieldAgentHome.test.tsx]: Thanks @mushrafulhoque-dsi.  Done in [this](https://github.com/opencrvs/opencrvs-core/pull/2066/commits/50c61758c1b47792680056942b7b67b250d59072) commit

File: packages/client/src/applications/index.ts (hunk 2 above)
Comment (tahmidrahman-dsi) [reply packages/client/src/applications/index.ts:621]: Agreed, I am going to amend the approach. Thanks @mushrafulhoque-dsi

File: packages/client/src/applications/index.ts (hunk 1 above)
Comment (tahmidrahman-dsi) [reply packages/client/src/applications/index.ts]: Thanks, it has been reverted

**Review by mushrafulhoque-dsi — APPROVED** [review]
👍

**Pull request merged**